### ISSUE TITLE
Remaining time for a job keeps increasing on its own

### ISSUE THREAD

**Issue by zaidoonhad** [jhuckaby/Cronicle#90]
## Summary

When running a test without a timeout (timeout set to 0), the job progress tab shows the test with n/a at the beggining (which is expected) but then the remaining time suddenly starts to increase as time goes by ( 52 min, 1 hour, 3 hours, 7 hours, etc...)
## Steps to reproduce the problem
I set the timeout for a job to 0 and ran the job.

### Operating system and version?

14.04.5 LTS

### Node.js version?

10.4.1

### Cronicle software version?

0.8.17

### Are you using a multi-server setup, or just a single server?
Single Server
### Are you using the filesystem as back-end storage, or S3/Couchbase?
filesystem

### Can you reproduce the crash consistently?
yes, although it is not a crash.

## Log Excerpts

<img width="444" alt="screen shot 2018-08-02 at 11 48 08 am" src="https://user-images.githubusercontent.com/26227298/43595735-6a08998c-964b-11e8-802a-edb674807d94.png">
<img width="124" alt="screen shot 2018-08-02 at 11 49 00 am" src="https://user-images.githubusercontent.com/26227298/43595736-6a1bd948-964b-11e8-84cb-d2c4d3155e7c.png">
<img width="148" alt="screen shot 2018-08-02 at 11 49 12 am" src="https://user-images.githubusercontent.com/26227298/43595737-6a2952bc-964b-11e8-9846-0a1a011a0400.png">
<img width="167" alt="screen shot 2018-08-02 at 11 50 18 am" src="https://user-images.githubusercontent.com/26227298/43595739-6a3b8194-964b-11e8-95db-423d779a090d.png">
<img width="160" alt="screen shot 2018-08-02 at 11 52 53 am" src="https://user-images.githubusercontent.com/26227298/43595740-6a5043ea-964b-11e8-9a23-5560c4e22d2a.png">

**Comment by jhuckaby** [comment]
This is actually "as designed" behavior.  Cronicle will try to estimate the time remaining, based on the total time elapsed, and the current `progress` number.  Note that it will *only* provide an estimate if your job sends at least one non-zero `progress` JSON event:

```js
{ "progress": 0.5 }
```

Now, here is the thing.  If you send a progress event and then nothing happens, the "estimate" will climb upwards.  That's because Cronicle is seeing that the total time elapsed is increasing, but the progress is not.  So it recalculates its remaining time estimate.  This is similar to how a GPS nav system recalculates its arrival time estimate if you run into traffic on the highway.

See [Reporting Progress](https://github.com/jhuckaby/Cronicle#reporting-progress) in the docs for more details.

**Comment by zaidoonhad** [comment]
Thank you for your time, but the weird thing is that I never send a JSON event to update the progress and this behaviour didn't happen before upgrading to 0.8.17 (I was on 0.8.16)

**Comment by zaidoonhad** [comment]
I just figured out what was causing the issue, so it had nothing to do with upgrading versions. I am sending the contents of an xml file at the beginning of the job  and that's causing the progress to update. now the xml does not contain the work progress at all.

**Comment by jhuckaby** [comment]
Ah, that gives me a clue.  Are you using the [Shell Plugin](https://github.com/jhuckaby/Cronicle#built-in-shell-plugin) by any chance?  Because if so, that Plugin interprets your STDOUT output and looks for any plain integer number on a line, optionally followed by a percent (`%`) symbol.  If it finds this in your STDOUT, it assumes it's a progress update.  I'll bet your XML file contains something like that.  Just a plain number on a line would do it!

From the docs:

> You can still report intermediate progress with the Shell Plugin. It can accept JSON in the standard output format if enabled, but there is also a shorthand. You can echo a single number on its own line, from 0 to 100, optionally with a `%` suffix, and that will be interpreted as the current progress.

In retrospect I should have made the `%` symbol required, to avoid phantom progress updates like this.  Assuming that's what's happening anyway :)

**Comment by zaidoonhad** [comment]
yup, you are correct i have a number on one line, is their a workaround?

**Comment by jhuckaby** [comment]
I will release a patch tonight with a fix, but if you can't wait that long, and you feel a little adventurous, it's literally a single character change in one of the Cronicle source files.  Just edit [line 44 in shell-plugin.js](https://github.com/jhuckaby/Cronicle/blob/master/bin/shell-plugin.js#L44).  It is located on your Cronicle server(s) in this location:

```
/opt/cronicle/bin/shell-plugin.js
```

On line 44, change this:

```js
if (line.match(/^\s*(\d+)\%?\s*$/)) {
```

To this:

```js
if (line.match(/^\s*(\d+)\%\s*$/)) {
```

i.e. remove the question mark (`?`) just after the `\%`.  This means it'll only grab progress updates if the STDOUT line contains a number *followed by a percent sign*.  Previously the percent sign was optional.

You do not have to restart Cronicle, nor transpile anything.  This change will take effect immediately on the next job run.

Hope this helps, and I will release a patch tonight with a permanent fix.  I am re-opening this issue until I release the patch.

**Comment by zaidoonhad** [comment]
thank you for your help

**Comment by jhuckaby** [comment]
Should be fixed in [Version 0.8.19](https://github.com/jhuckaby/Cronicle/releases/tag/v0.8.19).  Thanks!
